### PR TITLE
Create imu launch pkg, setup and add Dockerfile

### PR DESCRIPTION
**Pull request by GreatAlexander**

### CODE REVIEW

File: av_imu_launch/config/av_gx5_15.yml
Comment (hect95) [inline av_imu_launch/config/av_gx5_15.yml:7]: Codespell hook fails due to "NED" word caught as typo:

`v_imu_launch/config/av_gx5_15.yml:7: NED ==> NEED, END, NOD`

Maybe we put explicitly the acronym "North-East-Down" ? Alternatively we can just ignore it in the .pre-commit-config.yaml

File: Dockerfile
Comment (hect95) [inline Dockerfile]: Given that this is a sensor driver, maybe we should explicitly state the pkg version just to avoid potential problems?

The docker gets build every time the `run.sh`, `dev.sh` and `build.sh` scripts are used. A potential new driver update will pass inadvertently specially after the `apt update`.

I run `apt list -a ros-humble-microstrain-inertial-driver` inside the docker and it seems the version you probably tested was `3.2.1-1jammy.20240304.151451`

File: av_imu_launch/package.xml
Comment (hect95) [inline av_imu_launch/package.xml:5]: Shall we define the package version as `1.0.0`? I just thought it would be a good idea for our reference in the future.

File: README.md
Comment (hect95) [inline README.md:12]: I think worth to specify here the IMU driver dep and version

File: av_imu_launch/CMakeLists.txt
Comment (hect95) [inline av_imu_launch/CMakeLists.txt]: I do not think we are going to define any c++ code in this pkg, we can simplify the CMakeLists.txt

File: README.md
Comment (hect95) [inline README.md:12]: Would be good if we can explain the docker scripts and the use of them (we can copy and paste this description across other AV repos).

As a context: For me when I see `build.sh` and `run.sh` along with a dockerfile, I immediately think that `build.sh` is for literally only building the docker and `run.sh` is to run the built image (and potentially bash CMD accepted as args). Instead these scripts always build the docker image up to a certain target - either `build` and `dev`.

File: run.sh
Comment (hect95) [inline run.sh]: I think this is very minor but, potentially, `run.sh` can accept an arg that specifies `dev` or `runtime` docker run.

Also, do we want to build the docker image every time we run the container? I think we can save time if we do not, specially if the amount of docker containers that defined our AV stack increases.

Let me know your comments on this one.

**Review by hect95 — COMMENTED** [review]
Leaving few minor comments :)

File: av_imu_launch/config/av_gx5_15.yml
Comment (GreatAlexander) [reply av_imu_launch/config/av_gx5_15.yml:7]: Indeed, the pre-commit (Which I got from [elsewhere](https://github.com/ros-controls/ros2_control/blob/master/.pre-commit-config.yaml)) is a little over-zealous right now.
I would add any original source code (e.g. from a fork) to be ignored by the .pre-commit (We're not trying to fix upstream source code, and we don't want to change it in our fork to avoid any conflicts). I forked a bunch of repos at the start for testing which may no longer need the source code, so that also needs doing.

For this particular pre-commit typo, I'll just reword the comment as you suggest. I would like to run pre-commit for all our work when applicable with few exceptions (e.g. I added [*.dae to be ignored](https://github.com/ipab-rad/car_description/pull/1/files#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9R131) in the car-description repo).

File: Dockerfile
Comment (GreatAlexander) [reply Dockerfile]: I flip-flopped about this a few times in the past, with pros/cons for pinning a specific version (In the past, we tended to pin specific versions _only_ iff no newer would do [since you assume no major interface breaks unless it's a major version change]). In fact recently (I think it was the spinnaker camera drivers) I was keen to get the newest driver asap because they were actively fixing an issue with the driver.

However, probably a better practice for big complex code-bases with multiple components [such as an AV stack] is to pin everything just in case, so I think this is how we should do it from now on. I'll add this pinned version (And we should do so from now on in all docker containers). I presume we should only pin hard dependencies like drivers & libraries, we can probably get away without pinning e.g. `nano` for the dev images.

File: av_imu_launch/package.xml
Comment (GreatAlexander) [reply av_imu_launch/package.xml:5]: Indeed, the very first time I was setting all of this up I just wanted to go with 0s until we made the first "release", which getting all of these initial PRs merged may qualify as.
Back in my day we had these version numbers automatically generated and populated given our changelog updates.
It was convenient, but I guess we don't have to go to such lengths of automation (yet).
There are multiple ways we could go about it, I don't think we will be publicising our work often, but it will be very convenient indeed to generate docker images which versions correspond to the pkg versions within them.

Some relevant reading:
https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes
https://medium.com/fiverr-engineering/major-minor-patch-a5298e2e1798
https://github.com/audrow/update-ros-pkg-versions

File: Dockerfile
Comment (hect95) [reply Dockerfile]: Agreed!

File: av_imu_launch/package.xml
Comment (hect95) [reply av_imu_launch/package.xml:5]: I was thinking about the same. It wouldn't hurt to just semi-automate something so we ensure proper versioning for the good practice

File: av_imu_launch/config/av_gx5_15.yml
Comment (GreatAlexander) [reply av_imu_launch/config/av_gx5_15.yml:7]: Changed my mind :sweat_smile: I'm adding it to the list of codespell ignored words since we "could" (Given it's an IMU launch repo) use the acronym NED with reasonable context and it will be picked up _again_.

File: av_imu_launch/CMakeLists.txt
Comment (GreatAlexander) [reply av_imu_launch/CMakeLists.txt]: Soz I'm a ROS2 newbie and a CMake noob and copy-pasted the wrong file, simplified now.

File: README.md
Comment (GreatAlexander) [reply README.md:12]: Roger roger

Comment (hect95) [reply README.md:12]: I have added an #Usage section

File: run.sh
Comment (hect95) [reply run.sh]: `runtime.sh` and `dev.sh` docker convenience scripts have been implemented.

**Comment by hect95** [comment]
@GreatAlexander This PR can be merged after testing on vehicle :)

File: av_imu_launch/package.xml
Comment (hect95) [inline av_imu_launch/package.xml:7]: Could you please add me as maintainer? :)

**Review by hect95 — CHANGES_REQUESTED** [review]

**Review by hect95 — DISMISSED** [review]
Good to go!

**Review by hect95 — APPROVED** [review]
Tested on vehicle so good to go!

**Pull request merged**
